polygeist/lib/polygeist/Passes/ConvertPolygeistToLLVM.cpp

```diff
@@ -823,6 +1339,17 @@ struct ConvertPolygeistToLLVMPass
       populateVectorToLLVMConversionPatterns(converter, patterns);
 
       converter.addConversion([&](async::TokenType type) { return type; });
+      // This overrides the default
+      if (useBarePtrCallConv)
+        converter.addConversion([&](MemRefType type) -> Optional<Type> {
+          if (!canBeLoweredToBarePtr(type))
+            return std::nullopt;
+          const auto elemType = converter.convertType(type.getElementType());
+          if (!elemType)
+            return Type{};
+          return LLVM::LLVMPointerType::get(elemType,
+                                            type.getMemorySpaceAsInt());
+        });
```

Comment:
Trick used in other code: conversions are tried in reverse order, so this overrides the default and tries the default when `canBeLoweredToBarePtr` is false.

Reply:
nice explanation, can you please add it as a comment in the code itself.

Reply:
Done!